### DISCUSSION TITLE
Can I pass multiple providers to a resource to allow deploying to two targets?

### DISCUSSION THREAD

**Discussion by sheldonhull** [pulumi/pulumi#8734] [Pulumi Providers] [answered]
In my scenario I'm trying to use the yaml renderer to generate the K8 deployment yaml, while also wanting to try and use Pulumi directly. 

Should this be possible or would it require me to duplicate the stack config with one being yaml and one being k8, which manages two state files separately based on stack?

It seems the `opts ...pulumi.ResourceOption` is for a varidiac interface parameter. Any examples on how this could be done? 

What I'm currently doing

```

if d.Onlyyaml {
	pterm.Info.Println("Onlyyaml set to true, so this won't deploy to k8 directly")
	renderProviderName = "k8s-yaml-renderer"
	pargs.RenderYamlToDirectory = pulumi.String("yaml")

} else {
	pargs.Kubeconfig = pulumi.String("/home/vscode/.kube/config")
}
renderProvider, err := kubernetes.NewProvider(ctx, renderProviderName, pargs)
```

What I'd like to be able to do:

```
providers := pulumi.ResourceOption{}
providers = append(providers, YAMLPROVIDER)

if !d.Onlyyaml {
providers = append(providers, KPROVIDER)
}
_, err = networkingv1.NewIngress(ctx, d.ServiceDeployName("ingress"), &networkingv1.IngressArgs{
...
}, providers)
```

Otherwise, I'm assuming I'd need to duplicate my stack config and treat the yaml like a new stack, one which deploys to the yaml in the directory and the other for deploying to k8 as a new environment with both having unique state files. 

Would be great to get some expertise and know what is the best route.

**Comment by rawkode** [comment, marked as answer]
Hey @sheldonhull,

`provider` in the `pulumi.CustomResourceOptions`  doesn't support taking a list of providers.

You could write a multiplexer function though:

```typescript
const multiplexer = (resource: pulumi.Resource): pulumi.Resource[] => {
  const resources: pulumi.Resource[] = [resource];

  const additionalProviders = [
    new k8s.Provider("provider-eu", {
        kubeconfig: "",
    });
  ];

  return [ resource, ...additionalProviders.map((provider): pulumi.Resource => Object.assign(resource, { opts: { provider, } }))];
}
```

Warning: I've not tested this code, but it looks OK 😅
